contracts/extensions/TokenSupplier.sol

```diff
@@ -0,0 +1,188 @@
+
+  /// @notice Called when upgrading the extension (currently a no-op)
+  function finishUpgrade() public override auth {}
+
+  /// @notice Called when deprecating (or undeprecating) the extension (currently a no-op)
+  function deprecate(bool _deprecated) public override auth {}
+
+  /// @notice Called when uninstalling the extension
+  function uninstall() public override auth {
+    selfdestruct(address(uint160(address(colony))));
+  }
+
+  /// @notice Initialise the extension, must be called before any tokens can be issued
+  /// @param _tokenSupplyCeiling Total amount of tokens to issue
+  /// @param _tokenIssuanceRate Number of tokens to issue per day
+  function initialise(uint256 _tokenSupplyCeiling, uint256 _tokenIssuanceRate) public onlyRoot {
+    require(lastPinged == 0, "token-supplier-already-initialised");
+
+    tokenSupplyCeiling = _tokenSupplyCeiling;
+    tokenIssuanceRate = _tokenIssuanceRate;
+    lastPinged = block.timestamp;
+    lastRateUpdate = block.timestamp;
+  }
+
+  /// @notice Update the tokenSupplyCeiling, cannot set below current tokenSupply
+  /// @param _tokenSupplyCeiling Total amount of tokens to issue
+  function setTokenSupplyCeiling(uint256 _tokenSupplyCeiling) public initialised onlyRoot onlyContract {
+    try ColonyExtension(msg.sender).identifier() returns (bytes32 extensionId) {
+      IColonyNetwork colonyNetwork = IColonyNetwork(colony.getColonyNetwork());
+      address installation = colonyNetwork.getExtensionInstallation(extensionId, address(colony));
+      require(installation == msg.sender, "token-supplier-not-managed-extension");
+      require(extensionId == VOTING_HYBRID, "token-supplier-cannot-set-value");
+    } catch {
+      require(false, "token-supplier-no-identifier");
+    }
+
+    tokenSupplyCeiling = _tokenSupplyCeiling;
+
+    emit TokenSupplyCeilingSet(tokenSupplyCeiling);
+  }
+
+  /// @notice Update the tokenIssuanceRate
+  /// @param _tokenIssuanceRate Number of tokens to issue per day
+  function setTokenIssuanceRate(uint256 _tokenIssuanceRate) public initialised onlyRoot onlyContract {
+    try ColonyExtension(msg.sender).identifier() returns (bytes32 extensionId) {
+      IColonyNetwork colonyNetwork = IColonyNetwork(colony.getColonyNetwork());
+      address installation = colonyNetwork.getExtensionInstallation(extensionId, address(colony));
+      require(installation == msg.sender, "token-supplier-not-managed-extension");
+      require(
+        extensionId == VOTING_HYBRID || (
+          extensionId == VOTING_REPUTATION &&
+          block.timestamp - lastRateUpdate >= 4 weeks &&
+          _tokenIssuanceRate <= add(tokenIssuanceRate, tokenIssuanceRate / 10) &&
+          _tokenIssuanceRate >= sub(tokenIssuanceRate, tokenIssuanceRate / 10)
+        ),
+        "token-supplier-cannot-set-value"
+      );
+    } catch {
+      require(false, "token-supplier-no-identifier");
+    }
+
+    // Issue any outstanding tokens under the previous rate
+    // issueTokens();
```

Comment:
Erm?

Reply:
Ah, a relic from some debugging. I'll put it back.